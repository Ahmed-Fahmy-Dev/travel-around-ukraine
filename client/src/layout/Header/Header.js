import React, { useState } from 'react';
import AppBar from '@mui/material/AppBar';
import Toolbar from '@mui/material/Toolbar';
import Container from '@mui/material/Container';
import BurgerMenu from '../BurgerMenu/BurgerMenu';
import TabletMenu from '../TabletMenu/TabletMenu';
import LogoHeader from '../LogoHeader/LogoHeader';
import HeaderSearchFiled from '../SearchFiled/HeaderSearchFiled';




const Header = () => {

    const [scrollY, setScrollY] = useState(window.scrollY)
    document.addEventListener('scroll', () => setScrollY(window.scrollY))

    const appBarSX = scrollY > 400 ?
        { background: 'gray', transition: '0.3s' } :
        { background: 'none', transition: '0.3s' };

    const containerSX = {
        px: { xs: 4, sm: 7, md: 10, lg: 20 }
    }

    return (
        <AppBar
            position="fixed"
            sx={appBarSX} >
            <Container
                maxWidth="xl"
                sx={containerSX} >
                <Toolbar disableGutters  >
                    <LogoHeader />
                    <HeaderSearchFiled />
                    <BurgerMenu />
                    <TabletMenu />
                </Toolbar>
            </Container>
        </AppBar >
    )

}








export default Header;