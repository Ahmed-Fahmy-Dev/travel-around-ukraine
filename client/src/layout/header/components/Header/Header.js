/* eslint-disable  react/prop-types */
import React, { useState, useEffect } from 'react';
import { AppBar, Toolbar, Container, useScrollTrigger, Slide } from '@mui/material';
import { NavLink } from 'react-router-dom';
import { BurgerMenu, TabletMenu, HeaderSearchField } from '..';
import { LogOutModal } from '../../../../components';

const HideOnScroll = ({ children }) => {
  const trigger = useScrollTrigger();
  return (
    <Slide appear={false} direction="down" in={!trigger}>
      {children}
    </Slide>
  );
};

const LogoHeader = () => (
  <NavLink to="/" style={{ textDecoration: 'none' }}>
    <img src="/assets/images/logo/logo.png" alt="logo" width="65" height="50" />
  </NavLink>
);

const Header = () => {
  const [scrollY, setScrollY] = useState();

  useEffect(() => {
    const handleScrollY = () => setScrollY(window.scrollY);
    document.addEventListener('scroll', handleScrollY);

    return () => document.removeEventListener('scroll', handleScrollY);
  }, []);

  const appBarSX = {
    background: 'primary.main',
    boxShadow: scrollY > 400 ? null : 'none',
    py: { xs: 1, tablet: 1.5, laptop: 2 },
    opacity: 0.95,
    transition: '0.3s !important',
  };

  return (
    <HideOnScroll>
      <AppBar sx={appBarSX}>
        <Container>
          <Toolbar disableGutters>
            <LogoHeader />
            <HeaderSearchField />
            <BurgerMenu />
            <TabletMenu />
            <LogOutModal />
          </Toolbar>
        </Container>
      </AppBar>
    </HideOnScroll>
  );
};

export default Header;
