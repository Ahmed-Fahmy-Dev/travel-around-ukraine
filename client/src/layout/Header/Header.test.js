import { render } from '@testing-library/react';
import Header from './Header';
import { BrowserRouter } from 'react-router-dom';

describe('Render Header', () => {
  test('should Header render', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
