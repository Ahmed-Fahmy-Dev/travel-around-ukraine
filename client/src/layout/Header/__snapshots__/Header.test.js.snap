// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header should Header render 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-pm0sck-MuiPaper-root-MuiAppBar-root"
    data-testid="header"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
          style="text-decoration: none;"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-jcctx-MuiSvgIcon-root"
            data-testid="TravelExploreOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19.3 16.9c.4-.7.7-1.5.7-2.4 0-2.5-2-4.5-4.5-4.5S11 12 11 14.5s2 4.5 4.5 4.5c.9 0 1.7-.3 2.4-.7l3.2 3.2 1.4-1.4-3.2-3.2zm-3.8.1c-1.4 0-2.5-1.1-2.5-2.5s1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5-1.1 2.5-2.5 2.5zM12 20v2C6.48 22 2 17.52 2 12S6.48 2 12 2c4.84 0 8.87 3.44 9.8 8h-2.07c-.64-2.46-2.4-4.47-4.73-5.41V5c0 1.1-.9 2-2 2h-2v2c0 .55-.45 1-1 1H8v2h2v3H9l-4.79-4.79C4.08 10.79 4 11.38 4 12c0 4.41 3.59 8 8 8z"
            />
          </svg>
        </a>
        <div
          class="css-1frlucq"
        >
          <div
            class="css-1iz4iln"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-d7njfo-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary css-wjkppp-MuiInputBase-root"
          >
            <input
              aria-label="search"
              class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
              placeholder="Searchâ€¦"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiBox-root css-12lz99j"
        >
          <button
            aria-controls="menu-appbar"
            aria-haspopup="true"
            aria-label="appbar"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-gmtkzn-MuiButtonBase-root-MuiIconButton-root"
            data-testid="burgerBtn"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <nav
          class="MuiBox-root css-1cb06yv"
        >
          <a
            aria-current="page"
            class="MuiTypography-root MuiTypography-body1 css-14ylmoy-MuiTypography-root active"
            href="/"
            style="color: rgb(255, 199, 0);"
          >
            Home
          </a>
          <a
            class="MuiTypography-root MuiTypography-body1 css-14ylmoy-MuiTypography-root"
            href="/catalogue"
          >
            Catalogue
          </a>
          <a
            class="MuiTypography-root MuiTypography-body1 css-14ylmoy-MuiTypography-root"
            href="/cart"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ShoppingCartIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
          </a>
        </nav>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
